libraries/botbuilder-dialogs-adaptive-testing/src/testActions/userActivity.ts

```diff
@@ -61,7 +61,10 @@ export class UserActivity extends TestAction implements UserActivityConfiguratio
             activity.from = Object.assign({}, activity.from);
             activity.from.id = this.user;
             activity.from.name = this.user;
+        } else if (this.activity?.from && Object.keys(this.activity.from)) {
+            activity.from = Object.assign({}, this.activity.from);
```

Comment:
Why the `Object.assign`? Would a simple `activity.from = { ...this.activity.from }` suffice?

Reply:
I would *suspect* so, but [.NET clones](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/TestActions/UserActivity.cs#L74), so thought maybe I should do the same. Thoughts?

Reply:
Looks like this still slipped through.

Reply:
right. Fixed the two that you *didn't* mention 🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️🤦‍♂️